packages/windows/src/RPA/Windows/keywords/action.py

```diff
@@ -123,19 +124,15 @@ def middle_click(
         return self._mouse_click(locator, "MiddleClick", wait_time, timeout)
 
     def _mouse_click(self, element, click_type, wait_time, timeout):
-        if timeout:
-            auto.SetGlobalSearchTimeout(timeout)
         click_wait_time = wait_time or self.ctx.wait_time
-        try:
+        with self.set_timeout(timeout):
```

Comment:
Really nice use of context manager here and a text book place where to utilize it 👍 Makes code much more readable and you never forget to set the old value back.

Reply:
Thx! Also not to forget that the previous behaviour was bugged (so this is not just a structural refactor), as a keyword that was setting a specific timeout was calling another that was either setting/unsetting it again in some bad places and that leaved branches of logic unguarded by the correct desired timeout setting. (or guarded by the wrong timeout amount)